scijava/scijava-ops/src/test/java/org/scijava/param/JavadocParameterTest.java

```diff
@@ -110,6 +126,101 @@ public void testJavadocMethodIR() {
 		isSuitableGenericOpMethodInfo(info);
 	}
 	
+	@Test
+	public void testJavadocMethodI() {
+		Iterator<OpInfo> infos = ops.env().infos("test.javadoc.methodI").iterator();
+
+		OpInfo info = infos.next();
+		if (infos.hasNext()) {
+			Assert.fail("Multiple OpInfos with name \"test.javadoc.method\"");
+		}
+		isSuitableGenericOpMethodInfo(info);
+	}
+
+	/**
+	 * Tests javadoc scraping of mutable taglet
+	 * 
+	 * @mutable foo the i/o argument
+	 */
+	@OpMethod(names = "test.javadoc.methodInplaceI", type = Inplaces.Arity1.class)
+	public static void OpMethodInplaceI(List<String> foo) {
+		for (int i = 0; i < foo.size(); i++) {
+			foo.set(i, foo.get(i) + " foo");
+		}
+	}
+
+	/**
+	 * Tests javadoc scraping of mutable taglet
+	 * 
+	 * @dependency inplace the Op being wrapped
+	 * @mutable foo the i/o argument
+	 */
+	@OpMethod(names = "test.javadoc.methodDependency", type = Inplaces.Arity1.class)
+	public static void OpMethodInplaceI(@OpDependency(
+		name = "test.javadoc.methodInplaceI") Inplaces.Arity1<List<String>> inplace,
+		List<String> foo)
+	{
+		inplace.mutate(foo);
+	}
+
+	@Test
+	public void testJavadocMethodInplaceI() {
+		Iterator<OpInfo> infos = ops.env().infos("test.javadoc.methodInplaceI").iterator();
+
+		OpInfo info = infos.next();
+		if (infos.hasNext()) {
+			Assert.fail("Multiple OpInfos with name \"test.javadoc.method\"");
+		}
+
+		// assert input names
+		String[] inputNames = info.inputs().stream().map(m -> m.getKey()).toArray(
+			String[]::new);
+		Assert.assertArrayEquals(inputNames, new String[] { "foo" });
+
+		// assert input descriptions
+		String[] inputDescriptions = info.inputs().stream().map(m -> m
+			.getDescription()).toArray(String[]::new);
+		Assert.assertArrayEquals(inputDescriptions, new String[] {
+			"the i/o argument"});
+
+		// assert output name
+		String outputName = info.output().getKey();
+		Assert.assertEquals("foo", outputName);
+
+		// assert output description
+		String outputDescription = info.output().getDescription();
+		Assert.assertEquals("the i/o argument", outputDescription);
+	}
+	
+	@Test
+	public void testJavadocMethodInplaceWithDepedency() {
+		Iterator<OpInfo> infos = ops.env().infos("test.javadoc.methodDependency").iterator();
+
+		OpInfo info = infos.next();
+		if (infos.hasNext()) {
+			Assert.fail("Multiple OpInfos with name \"test.javadoc.methodDependency\"");
+		}
+
+		// assert input names
+		String[] inputNames = info.inputs().stream().map(m -> m.getKey()).toArray(
+			String[]::new);
+		Assert.assertArrayEquals(inputNames, new String[] { "foo" });

```

Comment:
Expected and actual reversed... probably we should go through all the tests (😬) to double check this. Maybe we can write some heuristics to semi-automate it.

Reply:
Fixed with deaa0497

I don't know what kind of heuristics would be best here. Expected vs. actual seems more like a feel thing. I suppose we could look for `new` keywords or `String` literals? I would assume that this is a mostly-manual procedure though. Let's solve it in its own branch (this is noted in scijava/scijava#76)